crates/turbopack-css/src/references/import.rs

```diff
@@ -344,13 +218,40 @@ impl ImportAssetReference {
 #[turbo_tasks::value_impl]
 impl ModuleReference for ImportAssetReference {
     #[turbo_tasks::function]
-    fn resolve_reference(&self) -> Vc<ModuleResolveResult> {
-        css_resolve(
+    async fn resolve_reference(&self) -> Result<Vc<ModuleResolveResult>> {
+        let css_asset = Vc::try_resolve_downcast_type::<CssModuleAsset>(self.origin).await?;
+
+        let import_context = {
+            let own_attrs = (&*self.attributes.await?).as_reference_import_attributes();
+            let mut import_context: ImportContext = if let Some(my_css_asset) = css_asset {
+                if let Some(import_context) = (&*my_css_asset.await?).import_context {
+                    (*import_context.await?).to_owned()
+                } else {
+                    Default::default()
+                }
+            } else {
+                Default::default()
+            };
+
+            if let Some(layer) = own_attrs.layer {
+                import_context.layers.push(layer);
+            };
+            if let Some(supports) = own_attrs.supports {
+                import_context.supports.push(supports);
+            };
+            if let Some(media) = own_attrs.media {
+                import_context.media.push(media);
```

Comment:
I feel like we should use an IndexSet instead of a Vec here to deduplicates media, supports and layers